components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -242,11 +247,24 @@ public void removeExpiredSessionData(Timestamp timestamp) {
             statement = connection.prepareStatement(sqlDeleteTask);
             statement.setTimestamp(1, timestamp);
             statement.execute();
-            connection.commit();
-        } catch (SQLException | IdentityException e) {
-            log.error("Error while removing Session Data.", e);
+            if (!connection.getAutoCommit()) {
+                connection.commit();
+            }
+        } catch (SQLException e) {
+            log.error("Error while removing Session Data ", e);
```

Comment:
if a connection is used to process some database resource transactionally, conn.rollback() must be called in an event of an error.